indexing-service/src/main/java/io/druid/indexing/common/task/ArchiveTask.java

```diff
@@ -91,12 +98,15 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
       log.info("OK to archive segment: %s", unusedSegment.getIdentifier());
     }
 
-    // Move segments
-    for (DataSegment segment : unusedSegments) {
-      final DataSegment archivedSegment = toolbox.getDataSegmentArchiver().archive(segment);
-      toolbox.getTaskActionClient().submit(new SegmentMetadataUpdateAction(ImmutableSet.of(archivedSegment)));
+    if (toolbox.getTaskActionClient().submit(new LockUpgradeAction(getInterval()))) {
+      // Move segments
+      for (DataSegment segment : unusedSegments) {
+        final DataSegment archivedSegment = toolbox.getDataSegmentArchiver().archive(segment);
```

Comment:
Well, we should always retry if there is a hope of things working out, i.e. if we have not already been preempted.